moldesign/molecules/atomcollections.py

```diff
@@ -433,6 +434,50 @@ def bonds_to(self, other):
                 bonds.append(bond)
         return bonds
 
+    def align(self, other):
+        """ Aligns this list of atoms to another list of atoms.
+
+        Args:
+            other (AtomContainer): The list of atoms to align to. 
+
+        Returns:
+            rmsd_error(Float): The root mean square deviation measuring the alignment error.
+            xform(np.ndarray): The 4x4 matrix that transforms this list of atoms to the input list of atoms.
+        """
+        if len(self.atoms) != len(other.atoms):
+            raise ValueError('The number of atoms for fitting this atom list %d does not equal the number of atoms in the input list %d' % (len(self.atoms), len(other.atoms)))
+
+        # Create atom coordinate arrays.
+        num_atoms = len(self.atoms)
+        self_coords = np.zeros((num_atoms, 3), dtype=float)
+        other_coords = np.zeros((num_atoms, 3), dtype=float)
+        for i in range(0,num_atoms):
+            self_coords[i] = self.atoms[i].position.value_in(u.angstrom)
+            other_coords[i] = other.atoms[i].position.value_in(u.angstrom)
+
+        # Calculate the 4x4 matrix transforming self_coords into other_coords.
+        rmsd_error, xform = rmsd_align(self_coords, other_coords)
+        return rmsd_error, xform 
```

Comment:
This method should apply the transformation before returning. Three cases to handle:

1. if all the atoms in this container are part of the same molecule, then transform the entire molecule:
```python
self.atoms[0].molecule.transform(xform)
```
2. if all the atoms in this container don't belong to any molecule, just transform the atoms in this container
```python
self.transform(xform)
```
3. if all the atoms in this container belong to different molecules, `raise ValueError`

Reply:
I don't think this method should apply the transformation because what it transforms is based on the contents of the atom lists, a really significant operation, transforming the entire molecule, need to be aware of what your atom lists contain.

I don't see any reason to restrict the container atoms to a single molecule either, could be useful to position a group of molecules.

Reply:
Definitely see your point here. In terms of API philosophy, I still like the idea of a one-line method to automatically align an entire system, but we definitely need something that just calculates a transformation without applying it. 

Before going further, let's meet up (offline) and write out some API use cases to see what makes sense.

Reply:
OK, let's discuss offline.